### PR TITLE
Feat/create work information section UI

### PR DESCRIPTION
**Pull request by AniaPiWo**

### CODE REVIEW

**Review by NerdbordMentor — DISMISSED** [review]
## Code Review Report

### Task goal
The task goal was to create a Work Information section in the Create a Profile page. The UI should be implemented as close as possible to the provided design.

### Code changes review

#### app/(profile)/create-profile/page.tsx
The WorkInformation component has been imported and added to the CreateProfilePage component. This change aligns with the task requirements.

#### src/components/CreateProfile/WorkInformation/WorkInformation.tsx
A new WorkInformation component has been created. It includes dropdowns for Position and Seniority, a text area for Tech Stack, and checkboxes for Employment Type. This aligns with the task requirements.

#### src/components/CreateProfile/WorkInformation/WorkInformations.module.scss
The corresponding styles for the WorkInformation component have been added. Without the design, it's hard to say if it matches exactly, but it seems to be in line with the task requirements.

#### src/components/Filters/Filters.tsx
The Filters component has been refactored to use a DropdownFilter component instead of a FilterButton component. This change doesn't seem to be directly related to the task requirements, but it doesn't contradict them either.

#### src/inputs/Dropdowns/DropdownBio/DropdownBio.tsx and src/inputs/Dropdowns/DropdownBio/DropdownBio.module.scss
A new DropdownBio component has been created. It seems to be a reusable dropdown component with a label, which could be used in the WorkInformation component. This aligns with the task requirements.

#### src/inputs/Dropdowns/DropdownFilter/DropdownFilter.tsx and src/inputs/Dropdowns/DropdownFilter/DropdownFilter.module.scss
The FilterButton component has been renamed to DropdownFilter and now includes a label. This change doesn't seem to be directly related to the task requirements, but it doesn't contradict them either.

#### Other changes
Other changes in files like app/(profile)/create-profile/page.module.scss, app/(profile)/my-profile/page.module.scss, app/globals.scss, src/components/Container/Container.module.scss, src/components/Container/Container.tsx, src/components/CreateProfile/PersonalInfo/PersonalInfo.tsx, and src/inputs/Switch/Switch.module.scss are minor formatting changes or unrelated to the task requirements.

### Conclusion
The code changes seem to cover the task requirements of creating a Work Information section in the Create a Profile page. However, without seeing the actual design or running the code, it's hard to say if the implementation matches the design exactly.

File: src/components/CreateProfile/WorkInformation/WorkInformations.module.scss [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+.right {
+  flex: 0 0 40%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  gap: 24px;
+}
+
+.lettersCount {
```
Comment (wawrzonnn) [inline src/components/CreateProfile/WorkInformation/WorkInformations.module.scss:37]: this class is duplicated

File: src/components/CreateProfile/WorkInformation/WorkInformation.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  ): void => {
+    setFilters((prevFilters) => ({
+      ...prevFilters,
+      [buttonType]: prevFilters[buttonType].includes(option)
+        ? prevFilters[buttonType].filter(
+            (selectedOption) => selectedOption !== option,
+          )
+        : [...prevFilters[buttonType], option],
+    }))
+    console.log(filters)
```
Comment (wawrzonnn) [inline src/components/CreateProfile/WorkInformation/WorkInformation.tsx:33]: remove console.logs

File: src/inputs/Dropdowns/DropdownBio/DropdownBio.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            )}
+            {arrow === 'IoIosArrowUp' ? <IoIosArrowUp /> : <IoIosArrowDown />}
+          </div>
+        </button>
+        {isDropdownActive && (
+          <div className={styles.dropdown}>
+            {options.map((option, index) => (
+              <label key={index} className={styles.dropdownInput}>
+                <div
+                  className={`${styles.checkbox} ${
```
Comment (wawrzonnn) [inline src/inputs/Dropdowns/DropdownBio/DropdownBio.tsx:73]: To dynamic classes, you should use the "bindclasses" library included in the project's documentation.

**Review by wawrzonnn — CHANGES_REQUESTED** [review]

**Review by wawrzonnn — APPROVED** [review]

File: src/components/CreateProfile/WorkInformation/WorkInformation.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
+'use client'
+import React, { useState } from 'react'
+import styles from './WorkInformations.module.scss'
+import { DropdownBio } from '@/inputs/Dropdowns/DropdownBio/DropdownBio'
+import TextArea from '@/inputs/TextArea/TextArea'
+import CheckboxInput from '@/inputs/Checkbox/Checkbox'
+
+const WorkInformation = () => {
+  const initialState = {
```
Comment (Sumick) [inline src/components/CreateProfile/WorkInformation/WorkInformation.tsx:9]: **nitpick:** We should move it out from component body, and type it properly (type object, don't do `as` something)

File: src/components/CreateProfile/WorkInformation/WorkInformation.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      employment: prevFilters.employment.includes(option)
+        ? prevFilters.employment.filter(
+            (selectedOption) => selectedOption !== option,
+          )
+        : [...prevFilters.employment, option],
+    }))
+    console.log(filters)
+  }
+
+  const filterLists = {
```
Comment (Sumick) [inline src/components/CreateProfile/WorkInformation/WorkInformation.tsx:47]: **nitpick:** Let's move it out from component body

File: src/components/CreateProfile/WorkInformation/WorkInformation.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  const handleEmploymentType = (option: string): void => {
+    setFilters((prevFilters) => ({
+      ...prevFilters,
+      employment: prevFilters.employment.includes(option)
+        ? prevFilters.employment.filter(
+            (selectedOption) => selectedOption !== option,
+          )
+        : [...prevFilters.employment, option],
+    }))
+    console.log(filters)
```
Comment (Sumick) [inline src/components/CreateProfile/WorkInformation/WorkInformation.tsx:44]: **nitpick:** Log to remove

File: src/components/Filters/Filters.tsx [hunk 7]
```diff
@@ -1,91 +1,79 @@
...
-      case 'Seniority':
-        updateSelectedOptions(selectedSeniority, setSelectedSeniority)
-        break
-
-      case 'Availability':
-        updateSelectedOptions(selectedAvailability, setSelectedAvailability)
-        break
+const Filters: React.FC = () => {
+  const initialState = {
+    technology: [] as string[],
```
Comment (Sumick) [inline src/components/Filters/Filters.tsx:9]: **nitpick:** We should move it out from component body, and type it properly (type object, don't do `as` something)

File: src/components/Filters/Filters.tsx [hunk 8]
```diff
@@ -1,91 +1,79 @@
...
-    'MongoDB',
-  ]
-
-  const seniorityList = ['Intern', 'Junior', 'Mid', 'Senior', 'Lead / Expert']
-  const availabilityList = ['Full-time', 'Part-time', 'Contract']
-  const locationsList = ['Poland', 'Europe', 'Other']
+  const filterLists = {
+    technology: [
+      'Javascript',
+      'Python',
```
Comment (Sumick) [inline src/components/Filters/Filters.tsx:34]: **nitpick:** We should move it out from component body

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks good, but we need to do cleanup here before merging

**Review by DanielSzczucki — APPROVED** [review]

**Pull request merged**